presentation/src/main/java/com/anytypeio/anytype/presentation/editor/EditorViewModel.kt

```diff
@@ -1800,6 +1803,46 @@ class EditorViewModel(
         proceedWithUpdatingActionsForCurrentSelection()
     }
 
+    private fun isNotAllBlocksSelected(): Boolean {
+        val allBlocks = orchestrator.stores.views.current().filter { it is BlockView.Selectable }
+        val selected = currentSelection()
+        return allBlocks.any { !selected.contains(it.id) }
+    }
+
+    private suspend fun updateSelectionUI(views: List<BlockView>) {
+        val targets = views.map { it.id }.toSet()
```

Comment:
all blocks are supposed to have unique IDS; why do you prefer to convert it to a set ?